stetho/src/main/java/com/facebook/stetho/inspector/elements/android/AndroidDOMProviderFactory.java

```diff
@@ -11,18 +11,59 @@
 
 import android.app.Application;
 
+import android.os.Handler;
+import android.os.Looper;
+import com.facebook.stetho.common.ThreadBound;
+import com.facebook.stetho.common.UncheckedCallable;
 import com.facebook.stetho.common.Util;
-import com.facebook.stetho.inspector.elements.DOMProvider;
+import com.facebook.stetho.common.android.HandlerUtil;
+import com.facebook.stetho.inspector.elements.DocumentProvider;
 
-public class AndroidDOMProviderFactory implements DOMProvider.Factory {
+public final class AndroidDOMProviderFactory
+    implements DocumentProvider.Factory, ThreadBound {
   private final Application mApplication;
+  private final Handler mHandler;
 
   public AndroidDOMProviderFactory(Application application) {
     mApplication = Util.throwIfNull(application);
+    mHandler = new Handler(Looper.getMainLooper());
   }
 
   @Override
-  public DOMProvider create() {
-    return new AndroidDOMProvider(mApplication);
+  public DocumentProvider create() {
+    return new AndroidDocumentProvider(mApplication, mHandler);
```

Comment:
hmm. If `AndroidDocumentProvider` only uses the `Handler` for the sake of implementing `ThreadBound`, then we can probably just pass `this` as the 2nd parameter, which could be re-typed to be of type `ThreadBound`.

```
public AndroidDocumentProvider(Application application, ThreadBound threadBound) { ...
```

This would then allow `AndroidDocumentProvider` to derive from your new `ThreadBoundProxy`.